examples/src/main/java/io/dapr/examples/invoke/grpc/HelloWorldClient.java

```diff
@@ -45,22 +41,11 @@ public class HelloWorldClient {
    * @param args Array of messages to be sent.
    */
   public static void main(String[] args) throws Exception {
-
     String user = "World";
-    String target = "localhost:" + System.getenv("DAPR_GRPC_PORT");
-
-    ManagedChannel channel = Grpc.newChannelBuilder(target, InsecureChannelCredentials.create())
-        .build();
-
-    try {
-      HelloWorldGrpc.HelloWorldBlockingStub blockingStub = HelloWorldGrpc.newBlockingStub(channel);
-
-      Metadata headers = new Metadata();
-      headers.put(Metadata.Key.of("dapr-app-id", Metadata.ASCII_STRING_MARSHALLER),
-          "hellogrpc");
-
-      // MetadataUtils.attachHeaders is deprecated.
-      blockingStub = blockingStub.withInterceptors(MetadataUtils.newAttachHeadersInterceptor(headers));
+    try (DaprClient client = new DaprClientBuilder().build()) {
+      HelloWorldGrpc.HelloWorldBlockingStub blockingStub = client.newGrpcStub(
+          "hellogrpc",
+          channel -> HelloWorldGrpc.newBlockingStub(channel));
```

Comment:
This is the new(est) design to do service invocation with gRPC proxying in Java.

Reply:
@artursouza I am not sure to understand this, this is just for invoking another service right? Do we need to expose the GrpcStub to the user? can we just not hide that behind the `invoke()` method? Sorry, I feel like I am missing something important and a big design decision here.

Reply:
We cannot because, like in this example, we don't have any idea about their service interface (HelloWorldGrpc in this case). Technically, the user does not even need to use our SDK for it, we are just making it easy for him to reuse the gRPC channel and the metadata interceptors behind the scenes.